bootstrap.sh

```diff
@@ -246,6 +246,16 @@ if [ -z "${CMAKE_COMMAND}" ] ||
   get_cmake_version
 fi
 
+# RHEL8: If cmake3 is too old, try cmake
+if [ "$CMAKE_COMMAND" = "cmake3" ] &&
+   [ "$CMAKE_MAJOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MAJOR" ] ||
+   [ "$CMAKE_MINOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MINOR" ] ||
+   [ "$CMAKE_REVISION" -lt "$CMAKE_GLOBAL_MIN_VERSION_REVISION" ] &&
```

Comment:
I think there are 2 problems with this:
- Due to the operator precedence the operands are evaluated in this order `([ "$CMAKE_COMMAND" = "cmake3" ] && [ "$CMAKE_MAJOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MAJOR" ]) || [ "$CMAKE_MINOR" -lt "$CMAKE_GLOBAL_MIN_VERSION_MINOR" ] || ([ "$CMAKE_REVISION" -lt "$CMAKE_GLOBAL_MIN_VERSION_REVISION" ] && [ -x "$(command -v cmake)" ])` while all the parts of the cmake version should be checked together
- I think the cmake version check doesn't work as it supposed to not just here but on line 235 and line 259 as well. For example with this check if the required version is 3.24.1 and our version is 3.25.0 it considers it lower than the required.

Reply:
both should be fixed now